pandas/core/common.py

```diff
@@ -78,17 +78,19 @@ def consensus_name_attr(objs):
     return name
 
 
-def maybe_box_datetimelike(value, dtype=None):
+def maybe_box_datetimelike(value, dtype=None, native=False):
```

Comment:
I don't like this function at all where it is. move to pandas/core/dtypes/cast.py

call it

maybe_box_to_native

and we don't need a native argument, the results as Timestamps are already correct.

Reply:
likely you would want to add coercsion for int/float/bool as these are numpy dtypes (but do that separately)

Reply:
> I don't like this function at all where it is. move to pandas/core/dtypes/cast.py
> 
> call it
> 
> maybe_box_to_native
> 
> and we don't need a native argument, the results as Timestamps are already correct.

Done both

Reply:
> likely you would want to add coercsion for int/float/bool as these are numpy dtypes (but do that separately)

Yes will do in a follow-on (to resolve #27616)